webpack.config.js

```diff
@@ -0,0 +1,71 @@
+const webpack = require("webpack");
+const path = require("path");
+const HtmlWebpackPlugin = require("html-webpack-plugin");
+const PnpWebpackPlugin = require("pnp-webpack-plugin");
+// Not used directly in playground, because it imports typescript
+const { SyntaxKind: LuaSyntaxKind } = require("typescript-to-lua/dist/LuaAST");
+
+const resolve = query => path.resolve(__dirname, query);
+
+/** @type {import("webpack").Configuration} */
+module.exports = {
+    devtool: "source-map",
+    entry: {
+        landing_bundle: resolve("src/landing/index.ts"),
+        play_bundle: resolve("src/playground/index.ts"),
+    },
+    output: { path: resolve("dist") },
+    node: { fs: "empty", module: "empty" },
+    resolve: {
+        extensions: [".tsx", ".ts", ".js"],
+        plugins: [PnpWebpackPlugin],
+        alias: {
+            // Replace vendored monaco-typescript services build with typescript, already used by typescript-to-lua
+            [require.resolve("monaco-editor/esm/vs/language/typescript/lib/typescriptServices.js")]: require.resolve(
+                "typescript",
+            ),
+
+            // Exclude builtin monaco-typescript libs
+            [require.resolve("monaco-editor/esm/vs/language/typescript/lib/lib.js")]: resolve("src/empty.ts"),
+        },
+    },
+    resolveLoader: {
+        plugins: [PnpWebpackPlugin.moduleLoader(module)],
+    },
+    module: {
+        rules: [
+            { test: /\.tsx?$/, use: "ts-loader" },
+            { test: /\.css$/, use: ["style-loader", "css-loader"] },
+            {
+                test: /\.(png|svg|jpg|gif|ico)$/,
+                loader: "url-loader",
+                options: { esModule: false, name: "[path][name].[ext]?hash=[hash:20]", limit: 8192 },
+            },
+            { test: /\.ttf$/, loader: "file-loader" },
+            { test: /\.scss$/, use: ["style-loader", "css-loader", "sass-loader"] },
+            { test: /\.webmanifest?$/, loader: "file-loader", options: { esModule: false } },
+        ],
+    },
+    plugins: [
+        new HtmlWebpackPlugin({
+            title: "TypeScriptToLua",
+            chunks: ["landing_bundle"],
+            template: resolve("assets/layout/template.html"),
+            filename: "index.html",
+            contentFile: "landing.html",
+        }),
```

Comment:
I'm not familiar really that familiar with webpack, could you clarify that this does only include the files required by the landing page and not Monoco, typescript, tstl ....

Because the initial reason why I split it into 2 projects/webpackconfigs was that the playground is massive in size (around 10mb irc) and I didn't want that overhead for a simple landing page.

Reply:
> could you clarify that this does only include the files required by the landing page and not Monoco, typescript, tstl ....

Yes, I have specified 2 different entry points and added 2 instances of `HtmlWebpackPlugin` with a `chunks` option specified, so the landing html page gets only modules referenced in `src/landing/index.ts`

Reply:
Awesome, I wish I had none that a year ago. Would have saved me a lot of work...